### DISCUSSION TITLE
Slow FPS despite little CPU and GPU utilization

### DISCUSSION THREAD

**Discussion by Josh015** [bevyengine/bevy#1410] [Q&A] [answered]
**Background**
About 10 years ago I made a simple demo program called ["Cubism"](https://vimeo.com/15442169) that utilized lots of cubes and lights as programmer art for my first big C++ OpenGL project. This week I decided to take Rust and Bevy for a spin by rewriting it with them as the new foundation.

**Problem**
I'm not sure what I'm doing wrong but it's SUPER sluggish. According to bevy's diagnostics plugins I'm averaging ~8fps when I run it. Though it's weird because task manager shows minimal CPU and GPU usage when the program is running. Also, I currently only have one light in the scene, and all the cube animations are pretty simple despite the large quantity of them on screen. 

**Question**
Is anyone willing to take a look to see if I've made a noob mistake and did something obvious that's killing my performance? If so, the full ["cubism-demo-rs"](https://github.com/Josh015/cubism-demo-rs) project is available online and doesn't require any special work to get running.

Thanks.

**Comment by bjorn3** [comment]
bevy_render currently is very much unoptimized. I think the low CPU and GPU usage mean that a lot of time is spent waiting on CPU to GPU transfers to finish.

**Comment by cart** [comment, marked as answer]
Are you building in release mode? I get over 200 fps on `cubism-demo-rs`

```cargo run --release```

And as @bjorn3 said, theres still plenty of optimization potential in Bevy.

**Reply by zicklag** [reply]
I get full smooth at least 60 fps ( I didn't measure ) when cubism in release mode on Linux.

**Reply by Josh015** [reply]
Hey, I finally got back here to see this. No, I wasn't doing "release" mode before, and yes it does make a shocking difference in performance.

Thanks for the tip.

**Reply by Josh015** [reply]
@cart  FYI, thus far the porting process has been wonderful. Even in this early stage I absolutely love working with Bevy! 😁
